### PR TITLE
[MAN-42] Add 'preboot' on heroku deployment.

### PR DESCRIPTION
**Pull request by JoseSolana**
## Summary: [MAN-42](https://loyal-guru.atlassian.net/browse/MAN-42)

Heroku deploys makes stuck the system a few seconds.
To try to solve it, [_preboot_](https://devcenter.heroku.com/articles/preboot) is being added.

A new label should be added to recognize  PR with migrations due to a [Heroku warning](https://devcenter.heroku.com/articles/preboot#deploying-with-preboot)
> If you make database schema changes that require downtime, we recommend disabling the preboot feature, performing the changes and code push as usual, then re-enabling preboot.

### Actions

- Add a new label
    - Label: `migration`
- PR with the `migration` label will be executed without _preboot_.
   > When the `migration` label is present, it'll be disabled before and be re-enabled after the deploy.
- _preboot_ will be always active

***

> Related: [[MANAGEMENT] 'HEROKU_API_KEY' set as environment to use Heroku CLI ](https://github.com/loyalguru/loyal-guru-api/pull/3683)

### CODE REVIEW

File: heroku_deploy/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine/git
+FROM node:alpine
```
Comment (JoseSolana) [inline heroku_deploy/Dockerfile:1]: Needed due to `RUN curl https://cli-assets.heroku.com/install.sh | sh` fails to find "node".

***

image

**Review by darccio — APPROVED** [review]

File: heroku_deploy/entrypoint.sh [hunk 2]
```diff
@@ -69,8 +69,19 @@ main(){
...
-  
-  git push https://heroku:${INPUT_HEROKU_API_KEY}@git.heroku.com/${app_name}.git HEAD:master -f
+
+  # The preboot will be always active, only when the label “migration” is present will be disabled.
+  # TODO: Once tested, 'staging_3' must be chenged bye 'production'
+  if  [ "${DEPLOY_ENVIRONMENT}" = "staging_3" ] && [ "${WITH_MIGRATION}" = "Y" ]; then
+    heroku features:disable preboot -a ${app_name}
+  fi
+
+  git push https://heroku:${HEROKU_API_KEY}@git.heroku.com/${app_name}.git HEAD:master -f
```
Comment (pbaro) [inline heroku_deploy/entrypoint.sh:79]: INPUT_HEROKU_API_KEY is now HEROKU_API_KEY. is that right?

**Review by nesitor — APPROVED** [review]

File: heroku_deploy/entrypoint.sh [hunk 3]
```diff
@@ -69,8 +69,19 @@ main(){
     app_name=${INPUT_HEROKU_APP_NAME_STAGING_THREE}
 
   fi
-  
-  git push https://heroku:${INPUT_HEROKU_API_KEY}@git.heroku.com/${app_name}.git HEAD:master -f
+
+  # The preboot will be always active, only when the label “migration” is present will be disabled.
+  # TODO: Once tested, 'staging_3' must be chenged bye 'production'
```
Comment (pbaro) [inline heroku_deploy/entrypoint.sh]:
```suggestion
  # TODO: Once tested, 'staging_3' must be changed to 'production'
```
for a brief moment, the 'bye production' thing scared the shit out of me

**Review by pbaro — APPROVED** [review]

File: heroku_deploy/entrypoint.sh (hunk 2 above)
Comment (pbaro) [reply heroku_deploy/entrypoint.sh:79]: ok, guess it depends on the env/with declaration in the workflow yml

**Review by manuuzkudun — APPROVED** [review]

File: heroku_deploy/entrypoint.sh (hunk 2 above)
Comment (JoseSolana) [reply heroku_deploy/entrypoint.sh:79]: @pbaro Yes, your conclusion was correct.

File: heroku_deploy/entrypoint.sh (hunk 3 above)
Comment (JoseSolana) [reply heroku_deploy/entrypoint.sh]: OMG, this is embarrassing 🤣.
Thank you

**Review by gcarrenolg — APPROVED** [review]

**Comment by JoseSolana** [comment]
# Works as expected

## Without `migration` label

- [Not disabling preboot](https://github.com/loyalguru/loyal-guru-api/pull/3683/checks?check_run_id=712131201#step:8:30)
   Not disabling preboot

 - [Nor re-enabling  preboot](https://github.com/loyalguru/loyal-guru-api/pull/3683/checks?check_run_id=712131201#step:8:344)
   Nor re-enabling  preboot

## With `migration` label

- [Disabling preboot](https://github.com/loyalguru/loyal-guru-api/pull/3683/checks?check_run_id=709948090#step:8:30)
  Disabling preboot

- [Enabling preboot](https://github.com/loyalguru/loyal-guru-api/pull/3683/checks?check_run_id=709948090#step:8:354)
  Enabling preboot

**Pull request merged**